hibernate-core/src/test/java/org/hibernate/orm/test/associations/ExplicitIdentifierTest.java

```diff
@@ -0,0 +1,200 @@
+							session.getDialect() instanceof HSQLDialect ?
+									"select COLUMN_name from \"TABLE_`ExplicitIdentifierTest$PersonImplicit`\""
+									: "select COLUMN_name from TABLE_ExplicitIdentifierTest$PersonImplicit",
+							String.class )
+					.getSingleResult() );
+
+			if ( session.getDialect().getSequenceSupport().supportsSequences() ) {
+				assertEquals( 51, session.createNativeQuery(
+						getSqlStringImplicit( session.getDialect() ),
+						Integer.class ).getSingleResult() );
+			}
+			else {
+				assertEquals( 51, session.createNativeQuery(
+						"select next_val from TABLE_TABLE_ExplicitIdentifierTest$PersonImplicit_SEQ",
+						Integer.class ).getSingleResult() );
+			}
+		} );
+	}
+
+	@Test
+	void testExplicitIdentifier(SessionFactoryScope scope) {
+
+		// explicit named object don't have a prefix
+
+		scope.inTransaction( session -> {
+			PersonExplicit personExplicit = new PersonExplicit();
+			personExplicit.name = "explicit";
+			session.persist( personExplicit );
+		} );
+
+		scope.inTransaction( session -> {
+
+			assertEquals( "explicit", session.createNativeQuery(
+							"select name from PersonExplicit",
+							String.class )
+					.getSingleResult() );
+
+			if ( session.getDialect().getSequenceSupport().supportsSequences() ) {
+				assertEquals( 51, session.createNativeQuery(
+						getSqlStringExplicit( session.getDialect() ),
+						Integer.class ).getSingleResult() );
+			}
+			else {
+				assertEquals( 51, session.createNativeQuery(
+						"select next_val from person_explicit_seq",
+						Integer.class ).getSingleResult() );
+			}
+		} );
+	}
+
+	private String getSqlStringImplicit(Dialect dialect) {
+		if ( dialect instanceof MariaDBDialect || dialect instanceof SQLServerDialect ) {
+			return "select next value for SEQUENCE_TABLE_ExplicitIdentifierTest$PersonImplicit_SEQ";
+		}
+		else if ( dialect instanceof HSQLDialect ) {
+			return "call next value for \"SEQUENCE_`TABLE_`ExplicitIdentifierTest$PersonImplicit`_SEQ`\"";
+		}
+		else if ( dialect instanceof DB2Dialect ) {
+			return "values next value for SEQUENCE_TABLE_ExplicitIdentifierTest$PersonImplicit_SEQ";
+		}
+		else if ( dialect instanceof OracleDialect ) {
+			return "select SEQUENCE_TABLE_ExplicitIdentifierTest$PersonImplicit_SEQ.nextval";
+		}
+		return "select nextval('SEQUENCE_TABLE_ExplicitIdentifierTest$PersonImplicit_SEQ')";
+	}
+
+	private String getSqlStringExplicit(Dialect session) {
+		if ( session instanceof MariaDBDialect || session instanceof SQLServerDialect ) {
+			return "select next value for person_explicit_seq";
+		}
+		else if ( session instanceof HSQLDialect ) {
+			return "call next value for person_explicit_seq";
+		}
+		else if ( session instanceof DB2Dialect ) {
+			return "values next value for person_explicit_seq";
+		}
+		else if ( session instanceof OracleDialect ) {
+			return "select person_explicit_seq.nextval";
+		}
+		return "select nextval('person_explicit_seq')";
+	}
```

Comment:
Alternatively you could call `dialect.getSequenceSupport().getSequenceNextValString(sequenceName)`?